File: app/web/src/services/auth.service.ts
import request from './request';
import type { 
  ApiResponse, 
  LoginResponse, 
  LoginByPasswordDTO, 
  LoginBySmsDTO,
  RegisterByPasswordDTO,
  RegisterBySmsDTO,
  SendSmsCodeDTO
} from '../types';

export const authService = {
  // 密码登录
  loginByPassword(data: LoginByPasswordDTO): Promise<ApiResponse<LoginResponse>> {
    return request.post('/member/login', {
      ...data,
      loginType: 'password',
    });
  },

  // 验证码登录
  loginBySms(data: LoginBySmsDTO): Promise<ApiResponse<LoginResponse>> {
    return request.post('/member/login', {
      ...data,
      loginType: 'sms',
    });
  },

  // 密码注册
  registerByPassword(data: RegisterByPasswordDTO): Promise<ApiResponse<LoginResponse>> {
    return request.post('/member/register', {
      phone: data.phone,
      password: data.password,
      registerType: 'password',
    });
  },

  // 验证码注册
  registerBySms(data: RegisterBySmsDTO): Promise<ApiResponse<LoginResponse>> {
    return request.post('/member/register', {
      ...data,
      registerType: 'sms',
    });
  },

  // 发送验证码
  sendSmsCode(data: SendSmsCodeDTO): Promise<ApiResponse<{ message: string }>> {
    return request.post('/member/sms-code', data);
  },

  // 登出
  logout(): Promise<ApiResponse<{ message: string }>> {
    return request.post('/member/logout');
  },
};

File: app/ios/ZayumMember/ZayumMember/Services/AuthService.swift
import Foundation

@Observable
class AuthService {
    static let shared = AuthService()
    
    private let apiClient = APIClient.shared
    private let tokenStorage = TokenStorage.shared
    
    private init() {}
    
    // MARK: - Login
    
    func loginByPassword(phone: String, password: String) async throws -> Member {
        let request = LoginByPasswordRequest(phone: phone, password: password)
        let response: LoginResponse = try await apiClient.post(
            path: APIEndpoints.login,
            body: request
        )
        
        tokenStorage.saveToken(response.token.accessToken)
        return response.user
    }
    
    func loginBySms(phone: String, code: String) async throws -> Member {
        let request = LoginBySmsRequest(phone: phone, code: code)
        let response: LoginResponse = try await apiClient.post(
            path: APIEndpoints.login,
            body: request
        )
        
        tokenStorage.saveToken(response.token.accessToken)
        return response.user
    }
    
    // MARK: - Register
    
    func registerByPassword(phone: String, password: String, confirmPassword: String) async throws -> Member {
        let request = RegisterByPasswordRequest(
            phone: phone,
            password: password,
            confirmPassword: confirmPassword
        )
        let response: LoginResponse = try await apiClient.post(
            path: APIEndpoints.register,
            body: request
        )
        
        tokenStorage.saveToken(response.token.accessToken)
        return response.user
    }
    
    func registerBySms(phone: String, code: String, password: String, confirmPassword: String) async throws -> Member {
        let request = RegisterBySmsRequest(
            phone: phone,
            code: code,
            password: password,
            confirmPassword: confirmPassword
        )
        let response: LoginResponse = try await apiClient.post(
            path: APIEndpoints.register,
            body: request
        )
        
        tokenStorage.saveToken(response.token.accessToken)
        return response.user
    }
    
    // MARK: - SMS Code
    
    func sendSmsCode(phone: String, type: SmsCodeType) async throws {
        let request = SmsCodeRequest(phone: phone, type: type)
        let _: EmptyData = try await apiClient.post(
            path: APIEndpoints.smsCode,
            body: request
        )
    }
    
    // MARK: - Logout
    
    func logout() async throws {
        do {
            let _: EmptyData = try await apiClient.post(path: APIEndpoints.logout)
        } catch {
            // 即使请求失败也清除本地 token
        }
        tokenStorage.clearToken()
    }
}
